server.cpp

```diff
@@ -34,10 +34,18 @@ void connection::do_write() {
 	auto self(shared_from_this());
 
 	reply_.status = reply::ok;
-	reply_.headers[0].name = "Content-Length";
-	reply_.headers[0].value= std::to_string(reply_.content.size());
-	reply_.headers[1].name = "Content-Type";
-	reply_.headers[1].value = "text/plain";
+  header head0;
```

Comment:
Good catch, I forgot to instantiate the headers and assumed they were already created.